extensions/register.md

```diff
@@ -0,0 +1,157 @@
+---
+title: The REGISTER command
+layout: spec
+work-in-progress: true
+copyrights:
+  -
+    name: "Ed Kellett"
+    period: "2020"
+    email: "e@kellett.im"
+  -
+    name: "Val Lorentz"
+    period: "2021"
+    email: "progval+ircv3@progval.net"
+---
+
+
+## Motivation
+
+The lack of a standardized way to register an account has been a source of frustration for many participants
+for several years.
+
+This isn't the first attempt to fix it. Unlike the withdrawn `ACC` proposal, this effort aims to be more
+appealing to implementors by being severely restricted in scope. The only register-verify flows supported
+are those already in use in the wild. The authors hope that this stripped-down solution will serve as the
+basis for future organic expansion.
+
+### Capability
+
+This specification adds the `draft/register` capability, whose presence signifies that the server
+accepts the `REGISTER` command.
+
+The capability has an optional value, a comma-separated list of key-value pairs; the format is intended
+to follow the precedent set by [the multiline draft][multiline].
+
+The defined keys are:
+
+ * `before-connect` - if present, indicates the server supports early registration, so it
+   MUST NOT use `FAIL REGISTER COMPLETE_CONNECTION_REQUIRED`
+ * `email-required` - if present, registrations require a valid email address to process
+
+Clients MUST ignore any value assigned to these keys, and MUST ignore any unknown key.
+
+### Commands
+
+    REGISTER {<email> | "*"} <password>
```

Comment:
In case the server supports `before-connect`, what is the "curerent nick"? Should we require that a `NICK` command is sent before `REGISTER`?

Reply:
nice catch. something like this in the list of responses should be sufficient? and/or a line on the `before-connect` key description. it's possible in `before-connect` setups but given those are mostly an experimental use case tossed on top I'd rather not describe this in the 'mainline' spec path.

>     FAIL REGISTER NEED_NICK <message>
> 
> Sent by the server if the client sends `REGISTER` before sending a `NICK` command. This is only possible in setups that advertise the `before-connect` key.

Reply:
I added a `*` argument before `<message>`, for consistency with the other errors.